model/trigger/version_test.go

```diff
@@ -210,3 +233,58 @@ func (s *VersionSuite) TestVersionOutcome() {
 	s.NoError(err)
 	s.NotNil(n)
 }
+
+func (s *VersionSuite) TestVersionRegression() {
+	t1 := task.Task{
+		Id:                  "t1",
+		Version:             s.version.Id,
+		BuildVariant:        "test",
+		Project:             s.version.Branch,
+		DisplayName:         "Test",
+		Status:              evergreen.TaskSucceeded,
+		Requester:           evergreen.RepotrackerVersionRequester,
+		RevisionOrderNumber: 2,
+	}
+	s.NoError(t1.Insert())
+
+	// a successful version should not generate
+	s.data.Status = evergreen.VersionSucceeded
+	n, err := s.t.versionRegression(&s.subs[3])
+	s.NoError(err)
+	s.Nil(n)
+
+	// a failing version for the first time should generate
+	t2 := task.Task{
+		Id:                  "t2",
+		Version:             s.version.Id,
+		BuildVariant:        "test",
+		Project:             s.version.Branch,
+		DisplayName:         "Test",
+		Status:              evergreen.TaskFailed,
+		Requester:           evergreen.RepotrackerVersionRequester,
+		RevisionOrderNumber: 2,
+	}
+	s.NoError(t2.Insert())
+	s.data.Status = evergreen.VersionFailed
+	n, err = s.t.versionRegression(&s.subs[3])
+	s.NoError(err)
+	s.NotNil(n)
+
+	// a version with a task that has failed before should not generate
+	t3 := task.Task{
+		Id:                  "t3",
+		Version:             "foo",
```

Comment:
You do "task.ByVersion" in the trigger, so this document is never fetched from the database by the test.

Reply:
It's fetched when previous runs of the task are looked up

Reply:
Got it.